pkg/wallet/models.go

```diff
@@ -62,3 +60,35 @@ type DeriveOptions struct {
 	// Nonce to prove uniqueness or freshness of the proof.
 	Nonce string `json:"nonce,omitempty"`
 }
+
+// QueryByExampleDefinition is model for QueryByExample query type.
+// https://w3c-ccg.github.io/vp-request-spec/#query-by-example
+type QueryByExampleDefinition struct {
+	Example *ExampleDefinition `json:"example"`
+}
+
+// QueryByFrameDefinition is model for QueryByExample query type.
+// https://w3c-ccg.github.io/vp-request-spec/
+// TODO QueryByExampleDefinition model is not yet finalized - https://github.com/w3c-ccg/vp-request-spec/issues/8
+type QueryByFrameDefinition struct {
+	Frame         map[string]interface{}    `json:"frame"`
+	TrustedIssuer []TrustedIssuerDefinition `json:"trustedIssuer"`
+}
+
+// ExampleDefinition frame for QueryByExample & QueryByFrame query types.
+// Refer - https://w3c-ccg.github.io/vp-request-spec/#example-2-a-query-by-example-query
+// TODO currently `IssuerQuery` is ignored.
+type ExampleDefinition struct {
```

Comment:
Should the name be 'Example' here ? Looks like this supports both QueryByExample and QueryByFrame from above comments.

Reply:
Yes, `example` frame can be part of QueryByFrame, but this [issue](https://github.com/w3c-ccg/vp-request-spec/issues/8) still not closed (vp-request-spec is not yet updated) so didn't include it . Will remove `QueryByFrame` from comment